dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPUtils.java

```diff
@@ -239,6 +239,27 @@ private static ChannelFuture open(NettyFrameworkImpl framework, final Channel ch
 		});
 		return openFuture;
 	}
+	
+	private static ChannelFutureListener generateOpenListenerWrapper(NettyFrameworkImpl framework, ChannelFutureListener listener) {
+		return new ChannelFutureListener() {
+		    @Override
+		    public void operationComplete(ChannelFuture future) throws Exception {
+		        framework.getExecutorService().execute(new Runnable() {
+					
+					@Override
+					public void run() {
+						try {
+							listener.operationComplete(future);
+						} catch (Exception e) {
+							System.out.println("Exception caught running open listener!! Closing channel just in case");
+							e.printStackTrace();
```

Comment:
Debug here explaining that the connection is being closed would be helpful. Not sure that the stack needs to be shared, this early into the open cycle, but if it is, we can add it formally to the debug statements